barretenberg/cpp/src/barretenberg/dsl/acir_format/ec_operations.hpp

```diff
@@ -11,6 +11,24 @@
 
 namespace acir_format {
 
+/**
+ * @brief Constraints for addition of two points on the Grumpkin curve.
+ *
+ * @details EcAdd constraints have 10 components:
+ * - input1_x: x-coordinate of the first input point
+ * - input1_y: y-coordinate of the first input point
+ * - input1_infinite: flag indicating if the first input point is the point at infinity
+ * - input2_x: x-coordinate of the second input point
+ * - input2_y: y-coordinate of the second input point
+ * - input2_infinite: flag indicating if the second input point is the point at infinity
+ * - predicate: flag indicating whether the constraint is active
+ * - result_x: witness index for the x-coordinate of the resulting point
+ * - result_y: witness index for the y-coordinate of the resulting point
+ * - result_infinite: witness index for the flag indicating if the result is the point at infinity
+ *
+ * The data related to input1 and input2 can either be given by witnesses or constants. However, x and y coordinates
```

Comment:
nit: not just the x and y coordinates of the same input must be both witnesses or constants, but the infinity flag also should be consistent, update the comment. 
```suggestion
 * The data related to input1 and input2 can either be given by witnesses or constants. However, x and y coordinates, and the infinity flag,
```

Reply:
While Noir does pass either all constants or all witnesses (see [here](https://github.com/noir-lang/noir/blob/0de2ac86d6cd969e2774f1017b4f104c64bfc039/compiler/noirc_evaluator/src/acir/acir_context/black_box.rs#L157)) barretenberg can handle the case `x`, `y` witnesses and `infinite` constant. Decided to leave the comment as is so to reflect barretenberg's requirements.